data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/DefaultEventMetadata.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.model.event;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.time.Instant;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A Basic implementation of {@link EventMetadata} interfaces utilizing an immutable map for attributes.
+ *
+ * @since 1.2
+ */
+public class DefaultEventMetadata implements EventMetadata {
+
+    private final String eventType;
+
+    private final Instant timeReceived;
+
+    private final ImmutableMap<String, Object> attributes;
+
+    private DefaultEventMetadata(final Builder builder) {
+
+        checkNotNull(builder.eventType, "eventType cannot be null");
+        checkArgument(!builder.eventType.isEmpty(), "eventType cannot be empty");
+
+        this.eventType = builder.eventType;
+
+        this.timeReceived = builder.timeReceived == null ? Instant.now() : builder.timeReceived;
+
+        this.attributes = builder.attributes == null ? new ImmutableMap.Builder<String, Object>().build() : ImmutableMap.copyOf(builder.attributes);
+    }
+
+    @Override
+    public String getEventType() {
+        return eventType;
+    }
+
+    @Override
+    public Instant getTimeReceived() {
+        return timeReceived;
+    }
+
+    @Override
+    public Map<String, Object> getAttributes() {
+        return attributes;
+    }
+
+    /**
+     * A Builder for creating {@link DefaultEventMetadata} instances.
+     *
+     * @since 1.2
+     */
+    public static class Builder {
+        private String eventType;
+        private Instant timeReceived;
+        private Map<String, Object> attributes;
+
+        /**
+         * Sets the event type. A non-null or empty event type is required, otherwise it will cause {@link #build()} to fail.
+         * @param eventType the event type
+         * @since 1.2
+         */
+        public Builder withEventType(final String eventType) {
```

Comment:
Do we want eventType to be an enum value so that we clearly define the different types of events within Data Prepper?